crates/kornia-tensor/src/parent_deallocator.rs

```diff
@@ -0,0 +1,11 @@
+use std::panic::RefUnwindSafe;
+
+/// A trait for deallocating memory of parent for tensor.
+///
+/// # Safety
+///
+/// The parent deallocator must be thread-safe.
+pub trait ParentDeallocator: Send + Sync + RefUnwindSafe {
```

Comment:
Maybe using a generic instead of a virtual table would be better?

Reply:
Allocators are generics already in the tensor storage structure, if that’s what you meant